src/core/model/search.rs

```diff
@@ -42,7 +42,7 @@ impl IdSearchReq {
 ///
 /// This struct derives the `Debug` trait, enabling it to be formatted using the `{:?}` formatter
 /// for debugging purposes.
-#[derive(Debug, Clone, Copy)]
```

Comment:
`IdSearchRes`  is already fixed-size hence is already a `Copy`, no need to explicitly derive it.